src/api/createDeviceReferenceDataASActions/src/v1/steps.py

```diff
@@ -0,0 +1,142 @@
+from domain.core.device_reference_data.v1 import DeviceReferenceData
+from domain.core.error import ConfigurationError
+from domain.core.product_key.v1 import ProductKeyType
+from domain.core.questionnaire.v3 import Questionnaire, QuestionnaireResponse
+from domain.repository.device_reference_data_repository.v1 import (
+    DeviceReferenceDataRepository,
+)
+from domain.repository.errors import AlreadyExistsError
+from domain.repository.questionnaire_repository.v2 import QuestionnaireRepository
+from domain.repository.questionnaire_repository.v2.questionnaires import (
+    QuestionnaireInstance,
+)
+from domain.request_models.v1 import (
+    CreateDeviceReferenceAdditionalInteractionsDataParams,
+)
+from domain.response.validation_errors import mark_validation_errors_as_inbound
+
+DEVICE_NAME_MARKER = "AS Additional Interactions"
+
+
+@mark_validation_errors_as_inbound
+def parse_device_reference_data_for_epr_payload(
+    data, cache
+) -> CreateDeviceReferenceAdditionalInteractionsDataParams:
+    payload: dict = data[parse_event_body]
+    return CreateDeviceReferenceAdditionalInteractionsDataParams(**payload)
+
+
+def get_party_key(data, cache) -> str:
+    product: CpmProduct = data[read_product]
+    party_keys = [
+        key.key_value
+        for key in product.keys
+        if key.key_type is ProductKeyType.PARTY_KEY
+    ]
+    try:
+        (party_key,) = party_keys
+    except ValueError:
+        raise ConfigurationError(
+            "Cannot create Additional Interactions in Product without exactly one Party Key"
+        )
+    return party_key
+
+
+def require_no_existing_additional_interactions_device_reference_data(
+    data, cache
+) -> list[QuestionnaireResponse]:
+    product: CpmProduct = data[read_product]
+    repo = DeviceReferenceDataRepository(
+        table_name=cache["DYNAMODB_TABLE"], dynamodb_client=cache["DYNAMODB_CLIENT"]
+    )
+    results = repo.search(
+        product_team_id=product.product_team_id, product_id=product.id
+    )
+    if any(
+        device_reference_data.name.endswith(DEVICE_NAME_MARKER)
+        for device_reference_data in results
+    ):
+        raise AlreadyExistsError(
+            "Additional Interactions Device Reference Data already exists for this Product"
+        )
+
+
+def read_questionnaire(data, cache) -> Questionnaire:
+    return QuestionnaireRepository().read(
+        QuestionnaireInstance.SPINE_AS_ADDITIONAL_INTERACTIONS
+    )
+
+
+def validate_questionnaire_responses(data, cache) -> list[QuestionnaireResponse]:
+    questionnaire: Questionnaire = data[read_questionnaire]
+    payload: CreateDeviceReferenceAdditionalInteractionsDataParams = data[
+        parse_device_reference_data_for_epr_payload
+    ]
+    raw_questionnaire_responses = payload.questionnaire_responses[
+        QuestionnaireInstance.SPINE_AS_ADDITIONAL_INTERACTIONS
+    ]
+    return [questionnaire.validate(data=qr) for qr in raw_questionnaire_responses]
```

Comment:
Might be a silly question, but this will still work if there were no SPINE_AS_ADDITIONAL_INTERACTIONS questionnaire responses right?

Reply:
Can see this does work from the test below :)